streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -97,7 +98,7 @@ public void remove(final Bytes key) {
     @Override
     public void put(final Bytes key, final byte[] value) {
         final long segmentId = segments.segmentId(keySchema.segmentTimestamp(key));
-        final Segment segment = segments.getOrCreateSegment(segmentId, context);
+        final Segment segment = segments.getOrCreateSegmentIfLive(segmentId, context);
```

Comment:
nit: maybe add a debug level log4j indicating the put operation is skipped?

Reply:
I can do that. FWIW, we always skipped such records, I just changed the method name to reflect that fact.